pkg/networkserver/redis/registry.go

```diff
@@ -298,7 +301,7 @@ func getUplinkMatch(ctx context.Context, r redis.Cmdable, inputKeys, processingK
 				resetsFCnt = &pb.MACSettings.ResetsFCnt.Value
 			}
 			if pb.ApplicationIdentifiers != appID || pb.DeviceID != devID {
-				return nil, errDatabaseCorruption.New()
+				return nil, errDatabaseCorruption.WithCause(errInvalidIdentifiers.New())
```

Comment:
Maybe panic here too and remove the error completely ? Since you've already decided that these errors are unrecoverable anyway, might as well get the full information in every place available.

Reply:
I will do this a follow-up PR to proceed with merge asap

Reply:
Well, taking another look at this - it's not really a implementation error, so I don't think we should panic here. Theoretically there could be a corrupted device with such identifiers, but it should not stop us from trying to match other devices.
However, if we receive an unknown field from Redis, for example - that's when shit really hits the fan and the only reasonable thing to do is to crash.